AutomatedTesting/Gem/PythonTests/Prefab/tests/duplicate_prefab/DuplicatePrefab_ContainingNestedEntitiesAndNestedPrefabs.py

```diff
@@ -0,0 +1,110 @@
+                                                      CREATION_POSITION)
+    nested_entities_root_name = nested_entities_root.get_name()
+
+    # Creates new nested prefabs at the root level
+    # Asserts if creation didn't succeed
+    entity_to_nest = EditorEntity.create_editor_entity_at(CREATION_POSITION, name=NESTED_PREFABS_TEST_ENTITY_NAME)
+    assert entity_to_nest.id.IsValid(), "Couldn't create TestEntity"
+    entity_to_nest.add_component(PHYSX_COLLIDER_NAME)
+    assert entity_to_nest.has_component(PHYSX_COLLIDER_NAME), f"Failed to add a {PHYSX_COLLIDER_NAME}"
+    _, nested_prefab_instances = prefab_test_utils.create_linear_nested_prefabs(
+        [entity_to_nest], NESTED_PREFABS_FILE_NAME_PREFIX, NESTED_PREFABS_NAME_PREFIX, NUM_NESTED_PREFABS_LEVELS)
+    prefab_test_utils.validate_linear_nested_prefab_instances_hierarchy(nested_prefab_instances)
+
+    # Creates a new prefab containing the nested entities and the nested prefab instances
+    # Asserts if prefab creation doesn't succeed
+    _, new_prefab = Prefab.create_prefab(
+        [nested_entities_root, nested_prefab_instances[0].container_entity],
+        FILE_NAME_OF_PREFAB_WITH_NESTED_ENTITIES_AND_NESTED_PREFABS)
+    new_prefab_container_entity = new_prefab.container_entity
+
+    nested_entities_root_on_instance = new_prefab.get_direct_child_entities()[0]
+    nested_prefabs_root_container_entity_on_instance = new_prefab.get_direct_child_entities()[1]
+    if new_prefab.get_direct_child_entities()[0].get_name() != nested_entities_root_name:
+        nested_entities_root_on_instance, nested_prefabs_root_container_entity_on_instance = \
+            nested_prefabs_root_container_entity_on_instance, nested_entities_root_on_instance
+
+    assert nested_entities_root_on_instance.get_name() == nested_entities_root_name \
+           and nested_entities_root_on_instance.get_parent_id() == new_prefab_container_entity.id, \
+        f"The name of the first child entity of the new prefab '{new_prefab_container_entity.get_name()}' " \
+        f"should be '{nested_entities_root_name}', " \
+        f"not '{nested_entities_root_on_instance.get_name()}'"
+    prefab_test_utils.validate_linear_nested_entities(nested_entities_root_on_instance, NUM_NESTED_ENTITIES_LEVELS,
+                                                      CREATION_POSITION)
+
+    # Duplicates the prefab instance and asserts if duplication doesn't succeed
+    Prefab.duplicate_prefabs([new_prefab])
+
+    # Test undo/redo on prefab duplication
+    general.undo()
+    wait_for_propagation()
+    search_filter = entity.SearchFilter()
+    search_filter.names = [FILE_NAME_OF_PREFAB_WITH_NESTED_ENTITIES_AND_NESTED_PREFABS]
+    prefab_entities_found = len(entity.SearchBus(bus.Broadcast, 'SearchEntities', search_filter))
+    assert prefab_entities_found == 1, "Undo failed: Found duplicated prefab entities"
+    search_filter.names = ["Entity_2"]
```

Comment:
This only verifies that one of the children got duplicated correctly rather than verifying that the entire tree got duplicated. I suggest you move the [code here](https://github.com/o3de/o3de/blob/bb839a207dfd769ee812577f634da40f86df9fc1/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/prefab_utils.py#L283-L290) into a helper function and reuse it for all 3 actions('duplicate_prefabs', 'undo' and 'redo' )to perform the same checks